includes/data-port/class-sensei-data-port-manager.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * File containing the Sensei_Data_Port_Manager class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * This class is responsible creating, deleting and keeping track of data port jobs.
+ */
+class Sensei_Data_Port_Manager implements JsonSerializable {
+	const OPTION_NAME = 'sensei-tools-jobs';
+
+	/**
+	 * An array of all in progress data port jobs. It has the following format:
+	 * {
+	 *
+	 *     @type string $id        Unique id for this job.
+	 *     @type int    $user_id   The user which initiatied this job.
+	 *     @type int    $time      When the job started.
+	 *     @type string $handler   The class which handles this job.
+	 * }
+	 *
+	 * @var array
+	 */
+	private $data_port_jobs;
+
+	/**
+	 * Tracks if the data port jobs have been updated.
+	 *
+	 * @var boolean
+	 */
+	private $has_changed;
+
+	/**
+	 * Instance of singleton.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Sensei_Data_Port_Manager constructor.
+	 */
+	private function __construct() {
+		$json_string       = get_option( self::OPTION_NAME );
+		$this->has_changed = false;
+
+		if ( $json_string ) {
+			$this->data_port_jobs = json_decode( $json_string, true );
```

Comment:
Is there a reason to use JSON rather than just letting `get_option` and `update_option` serialize and deserialize for us?

Reply:
In general I think that JSON is a far better representation than PHP array serialization and it makes debugging easier. It also what we used in our latest enrolment work.

I am open to using PHP serialization if there are benefits with it and decide on it as common practice. @jake any ideas?

Reply:
I'm fine either way. Just curious whether JSON gives us enough of a tangible benefit, as it seems to add some extra work.

Reply:
I've been trying to use JSON for a while now, but mostly because I prefer it. While it isn't completely secure (p7H4VZ-2cf-wp), I think it is a bit safer than PHP's serialization. There used to be a performance benefit, but doesn't seem like that's the case in PHP 7.